sympy/functions/special/delta_functions.py

```diff
@@ -159,6 +159,26 @@ def is_simple(self, x):
             return p.degree() == 1
         return False
 
+    def _eval_rewrite_as_Piecewise(self, arg):
+        """Represents DiracDelta in a Piecewise form
+
+           Examples
+           ========
+
+           >>> from sympy import DiracDelta, Piecewise, Symbol
+           >>> from sympy.abc import x, y
+           >>> x = Symbol('x', real=True)
+
+           >>> DiracDelta(x).rewrite(Piecewise)
+           Piecewise((undefined, Eq(x, 0)), (0, True))
+
+           >>> DiracDelta(x - 5).rewrite(Piecewise)
+           Piecewise((undefined, Eq(x - 5, 0)), (0, True))
+
+        """
+        if arg.is_real:
+            return Piecewise(('undefined', Eq(arg, 0)), (0, True))
```

Comment:
I'd say this should be `oo`. Neither is technically correct, because DiracDelta isn't really a function, but `oo` is the most common way this function is represented, and if you ask for a Piecewise representation, this is probably what you want. 

Also, don't use strings directly as input to SymPy functions (create a `Symbol` or `Dummy` first).